### ISSUE TITLE
Improve auto-scrolling to be less buggy

### ISSUE THREAD

**Issue by wagnerand** [jonathanKingston/sea-containers#18]
So, I tried a few vertical-tabbar webextensions, and this one is the best yet! Congrats.

What slows down my productivity a bit is automatic scrolling when you select a tab, so it is automatically at the bottom of the viewport. It makes quickly switching between tabs more effort because you have to manually scroll "back" first.

Would you be open to adding an option that disables that?

**Comment by jonathanKingston** [comment]
@wagnerand hey!

In short, I am happy to allow this as an option as I can see it being annoying (I need to add a convenient way to change options without being too intrusive on normal UX). However I wanted to unpack my thoughts a little.

So I tried to capture moments when it's most helpful to auto scroll and when it isn't. I assume you are using keyboard shortcuts perhaps?

I am also trying to add to Firefox the ability to use Ctrl+Tab shortcuts within a web extension too (I did a preliminary patch however it needs some work), I would keep it within the same container order, do you think this would also prevent the scrolling issue.

I'm also expecting post 57 Firefox there to be a way to disable the tab strip too, this might be an option also however again I can see this as being a main pain point currently if you were using that.

The main use case I was trying to capture was mainly switch to tab within the awesomebar. When you might be jumping far across a container or moving to another.

**Comment by wagnerand** [comment]
Hm, I'm not sure I follow.

This is ~~only~~ mostly relevant when you *don't* use keyboard shortcuts, since using them allows you to jump to tabs that might be out of your viewport. If you change tabs by mouse, you can only click on what you see.

When using the default tab-strip, it does not auto-scroll at all, as long as the selected tab is within the viewport.

**Comment by jonathanKingston** [comment]
@wagnerand yea this is a bug not an enhancement then. (Sorry super tired)

Should scroll and open the container accordion if crossing into a different container:
  - Tab strip
  - Keyboard shortcuts
  - Open a new tab

Scroll into view doesn't make sense
  - Clicking a tab within the sidebar

If this was the behaviour would you still need the option to disable it? (I'm trying to emulate the tab strip scrolling behaviour as much as possible).

Are there other cases I haven't thought about?

**Comment by wagnerand** [comment]
Sounds fine to me!

I think scrolling if crossing into a different container is fine, but might be unnecessary if the target is within the viewport already?

**Comment by jonathanKingston** [comment]
@wagnerand yeah my hunch was already doing that. Will do some more debugging, it was just added in quickly.

**Comment by wagnerand** [comment]
👍 Thank you!
